### PR TITLE
Fix NaN handling in ecma_op_same_value

### PR DESCRIPTION
**Pull request by ruben-ayrapetyan**
Related issue: #130

### CODE REVIEW

File: jerry-core/ecma/operations/ecma-conversion.cpp [hunk 1]
```diff
@@ -102,22 +102,30 @@ ecma_op_same_value (ecma_value_t x, /**< ecma-value */
   {
     return false;
   }
-
-  if (is_x_undefined
-      || is_x_null)
+  else if (is_x_undefined
```
Comment (egavrin) [inline jerry-core/ecma/operations/ecma-conversion.cpp]: Why multiline?

```cpp
else if (is_x_undefined || is_x_null)
```

File: jerry-core/ecma/operations/ecma-conversion.cpp [hunk 2]
```diff
@@ -102,22 +102,30 @@ ecma_op_same_value (ecma_value_t x, /**< ecma-value */
...
   {
     ecma_number_t *x_num_p = ecma_get_number_from_value (x);
     ecma_number_t *y_num_p = ecma_get_number_from_value (y);
 
-    if (ecma_number_is_nan (*x_num_p)
-        && ecma_number_is_nan (*y_num_p))
+    bool is_x_nan = ecma_number_is_nan (*x_num_p);
+    bool is_y_nan = ecma_number_is_nan (*y_num_p);
+
+    if (is_x_nan
```
Comment (egavrin) [inline jerry-core/ecma/operations/ecma-conversion.cpp]: likewise

**Comment by egavrin** [comment]
Except style, good to me

File: jerry-core/ecma/operations/ecma-conversion.cpp (hunk 1 above)
Comment (ruben-ayrapetyan) [inline jerry-core/ecma/operations/ecma-conversion.cpp]: I think that in the case, in multiline mode, as equal `is_x_` parts are closer, it is simpler to compare the condition parts, and so, simpler to read the code.

Comment (egavrin) [inline jerry-core/ecma/operations/ecma-conversion.cpp]: In that case it's gimmick.

Comment (ruben-ayrapetyan) [inline jerry-core/ecma/operations/ecma-conversion.cpp]: Why?

Comment (egavrin) [inline jerry-core/ecma/operations/ecma-conversion.cpp]: Condition is too simple to move it on other line. The only reason to use multiline condition - it doesn't fit into 120.

Comment (ruben-ayrapetyan) [inline jerry-core/ecma/operations/ecma-conversion.cpp]: Why shouldn't we make multiline conditions in other cases, if it improves readability?

Comment (egavrin) [inline jerry-core/ecma/operations/ecma-conversion.cpp]: Because they don't - it just confusing readers.

**Comment by zherczeg** [comment]
lgtm. I don't mind the condition splitting style as long as it is not a rule.

File: jerry-core/ecma/operations/ecma-conversion.cpp (hunk 1 above)
Comment (ruben-ayrapetyan) [inline jerry-core/ecma/operations/ecma-conversion.cpp]: // The question was discussed verbally.
// Agreed on combining very simple, similar conditions to one line, and splitting other on several lines.

Comment (ruben-ayrapetyan) [inline jerry-core/ecma/operations/ecma-conversion.cpp]: I'll update pull request.

**Comment by ruben-ayrapetyan** [comment]
@egavrin, pull request was updated. Please, check.

**Comment by egavrin** [comment]
`make push`

**Pull request merged**